cpp/src/arrow/array/data.h

```diff
@@ -434,6 +435,23 @@ struct ARROW_EXPORT ArraySpan {
     return GetValues<T>(i, this->offset);
   }
 
+  // Access a buffer's data as a span
```

Comment:
I think this function can benefit from a more ellaborate (in Doxygen format) docstring.

```suggestion
  /// \brief Access a buffer's data as a span
  ///
  /// \param i The buffer index
  /// \param length The required length (in number of typed values) of the requested span
  /// \pre i > 0
  /// \pre length <= the length of the buffer (in number of values) that's expected for this array type
  /// \return A span<const T> of the requested length
```